src/VisualStudio/Xaml/Impl/Implementation/XamlTextViewCreationListener.cs

```diff
@@ -0,0 +1,185 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Immutable;
+using System.ComponentModel.Composition;
+using System.Linq;
+using System.Runtime.InteropServices;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.Editor;
+using Microsoft.CodeAnalysis.Editor.Xaml;
+using Microsoft.VisualStudio.Editor;
+using Microsoft.VisualStudio.LanguageServices.Implementation.ProjectSystem;
+using Microsoft.VisualStudio.OLE.Interop;
+using Microsoft.VisualStudio.Shell;
+using Microsoft.VisualStudio.Shell.Interop;
+using Microsoft.VisualStudio.Text.Editor;
+using Microsoft.VisualStudio.TextManager.Interop;
+using Microsoft.VisualStudio.Utilities;
+
+namespace Microsoft.VisualStudio.LanguageServices.Xaml
+{
+    [Export(typeof(IVsTextViewCreationListener))]
+    [ContentType(ContentTypeNames.XamlContentType)]
+    [TextViewRole(PredefinedTextViewRoles.PrimaryDocument)]
+    [TextViewRole(PredefinedTextViewRoles.Document)]
+    internal sealed partial class XamlTextViewCreationListener : IVsTextViewCreationListener
```

Comment:
At some point we would like to track all XAML documents, opened, projected, or closed. At that point we would probably use the same msbuild-based mechanism as C#/VB. But for now this is probably good enough.